platform/mk2/hal/fat_sd_stm32/sdcard_device_stm32.c

```diff
@@ -26,3 +26,8 @@ void disk_init_hardware(void)
 {
 	SD_Init();
 }
+
+bool sdcard_device_card_present(void)
```

Comment:
IMHO I would implement this as an enum where the values are as follows:

```c
SD_PRESENT_UNKNOWN = -1,
SD_PRESENT_FALSE = 0,
SD_PRESENT_TRUE = 1,
```

So we can tell the difference in code and logs.  Minor but worth a thought.